sqlite3.go

```diff
@@ -1901,16 +1904,18 @@ func (s *SQLiteStmt) Close() error {
 		return nil
 	}
 	s.closed = true
-	if !s.c.dbConnOpen() {
+	conn := s.c
+	stmt := s.s
+	s.c = nil
+	s.s = nil
+	runtime.SetFinalizer(s, nil)
+	if !conn.dbConnOpen() {
```

Comment:
I know it's existing, but I think this check is a bug. The docs for `sqlite3_close_v2` say the following:
> If sqlite3_close_v2() is called with unfinalized prepared statements, unclosed BLOB handlers, and/or unfinished sqlite3_backups, it returns [SQLITE_OK](https://www.sqlite.org/rescode.html#ok) regardless, but instead of deallocating the database connection immediately, it marks the database connection as an unusable "zombie" and makes arrangements to automatically deallocate the database connection after all prepared statements are finalized, all BLOB handles are closed, and all backups have finished.

That means that refusing to call `sqlite3_finalize` because the database has already been closed will in fact prevent the database from getting properly deallocated.

Reply:
This makes sense, I'll remove the check for `dbConnOpen` and add some tests around this tomorrow.

Reply:
Removed the check if the db is closed and added a test.